src/main/java/edu/harvard/iq/dataverse/AuxiliaryFileServiceBean.java

```diff
@@ -119,6 +134,35 @@ public AuxiliaryFile lookupAuxiliaryFile(DataFile dataFile, String formatTag, St
             return null;
         }
     }
+    
+    @SuppressWarnings("unchecked")
```

Comment:
We only use this `@SuppressWarnings("unchecked")` a few times in the code base and (without looking deeply into it) it seems like a code smell. Do we need it? What problem does it solve? I commented it out locally and the app still compiled fine.

Reply:
The smell part is the broad use of things like` (List<'SomeEntityClass'>)query.getResultList();` with a query that doesn't specify the class type in the list. Adding the  `@SuppressWarnings("unchecked")` is an improvement - hopefully indicating that a developer has checked the query and knows that only the right class can be returned in the list since the compiler can't (as with ~all of the @Entity queries in Dataverse). In general, I think I add the `@SuppressWarnings("unchecked")` because Eclipse flags the warning in the UI. 

That said, the better approach is to use a TypedQuery<AuxiliaryFile> class to start with. I've changed the code I added and the other examples in this Bean that weren't type checking before.

Reply:
Looks good with the TypedQuery. Thanks.